### PR TITLE
Add toNev to NonEmptyList variants

### LINKED ISSUE #3422: Add toNev combinator in NonEmptyList
I wanted to get an instance of NonEmptyVector from a NonEmptyList and I noticed it was lacking that combinator.

Something like:

```
def toNev[B >: A](implicit order: Order[B]): NonEmptyVector[B] = 
    NonEmptyVector.fromVectorUnsafe(toList.toVector)
```

There might be more combinators missing.

### PR DESCRIPTION
**Pull request by saraiva132**
Closes #3422

### CODE REVIEW

File: core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala [hunk 1]
```diff
@@ -445,6 +445,21 @@ class NonEmptyLazyListOps[A](private val value: NonEmptyLazyList[A])
...
+   * scala> import cats.instances.int._
+   * scala> val nel = NonEmptyLazyList.fromLazyListPrepend(1, LazyList(2,3,4))
+   * scala> val expectedResult = NonEmptyVector.fromVectorUnsafe(Vector(1,2,3,4))
+   * scala> val result = nel.toNev
+   * scala> result === expectedResult
+   * res0: Boolean = true
+   *}}}
+   */
+  final def toNev[B >: A]: NonEmptyVector[B] =
+    NonEmptyVector.fromVectorUnsafe(toLazyList.toList.toVector)
```
Comment (LukaJCB) [inline core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala:461]: does `LazyList` not have its own `toVector` method?

Comment (saraiva132) [reply core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala:461]: You are right. It does. Additionally, I was missing an import. I will push changes. Thank you @LukaJCB :)

Comment (LukaJCB) [reply core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala]: Of course :)

File: core/src/main/scala/cats/data/NonEmptyList.scala [hunk 2]
```diff
@@ -437,6 +437,21 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) extends NonEmptyCollec
...
+   * scala> import cats.data._
+   * scala> import cats.instances.int._
+   * scala> val nel = NonEmptyList(1, List(2,3,4))
+   * scala> val expectedResult = NonEmptyVector.fromVectorUnsafe(Vector(1,2,3,4))
+   * scala> val result = nel.toNev
+   * scala> result === expectedResult
+   * res0: Boolean = true
+   *}}}
+   */
+  def toNev[B >: A]: NonEmptyVector[B] =
```
Comment (travisbrown) [inline core/src/main/scala/cats/data/NonEmptyList.scala:453]: Is there a reason this one isn't final but the one above is?

Comment (saraiva132) [reply core/src/main/scala/cats/data/NonEmptyList.scala:453]: Yes. NonEmptyList is a final class. LazyNonEmptyList is not.

**Pull request merged**